tests/test_class_basics.rs

```diff
@@ -244,3 +244,45 @@ fn panic_unsendable_base() {
 fn panic_unsendable_child() {
     test_unsendable::<UnsendableChild>().unwrap();
 }
+
+fn to_string(obj: &PyAny) -> PyResult<String> {
+    Ok(obj.str()?.to_str()?.to_string())
```

Comment:
```suggestion
    obj.extract()
```

Reply:
This one is tricky: I wanted to perform conversion explicitly via python's `str()` function, so that we could accept any object, like `dict`, for example. With `extract` we'll only be able to accept `FromPy<String>` objects and using `from_py_with` wouldn't make sense.

It's quite confusing, so I'll try to come up with a better example for this test that shows the need for a custom conversion function more clearly.

Reply:
Changed the test a bit, hope it's less confusing now